beacon_node/execution_layer/src/test_utils/mock_builder.rs

```diff
@@ -193,7 +180,7 @@ impl<E: EthSpec> MockBuilder<E> {
         beacon_url: SensitiveUrl,
         spec: ChainSpec,
         executor: TaskExecutor,
-    ) -> (Self, MockBuilderServer) {
+    ) -> (Self, (SocketAddr, impl Future<Output = ()>)) {
```

Comment:
same as above, we can just:
```suggestion
    ) -> (Self, SocketAddr) {
```
if we change the signature of the function to `async fn` and `.await` on `server`

Reply:
We want to lazily evaluate the `server` future though, after the socket address has been returned and used by the caller.

Reply:
Oh yea, this one I think we have to leave